corehq/apps/userreports/README.md

```diff
@@ -2005,8 +2005,33 @@ Following are some custom expressions that are currently available.
 
 You can find examples of these in [practical examples](examples/examples.md).
 
-## Inspecting database tables
+## Scaling UCR
+
+### Profiling data sources
+
+You can use `./manage.py profile_data_source <domain> <data source id> <doc id>`
+to profile a datasource on a particular doc. It will give you information such
+as functions that take the longest and number of database queries it initiates.
+
+### Faster Reporting
+
+If reports are slow, then you can add `create_index` to the data source to any
+columns that have filters applied to them.
```

Comment:
The biggest trade off is storage space though there is also some penalty for changes to the table since the index needs to be maintained.